scholar.go

```diff
@@ -321,8 +321,13 @@ func (sch *Scholar) QueryProfileWithMemoryCache(user string, limit int) ([]*Arti
 func (sch *Scholar) QueryProfileDumpResponse(user string, queryArticles bool, limit int, dumpResponse bool) ([]*Article, error) {
 	var articles []*Article
 
-	// todo: make page size configurable, also support getting more than one page of citations
-	requestURL := BaseURL + "/citations?user=" + user + "&cstart=0&pagesize=" + strconv.Itoa(limit)
+	// Use a reasonable page size (minimum 20 for Google Scholar) but respect the limit when processing
+	pageSize := limit
+	if pageSize < 20 {
```

Comment:
What happens if we have more than 20 articles in the profile? Do we have logic to request multiple pages if we're less than the limit still?